rosette/api.py

```diff
@@ -455,24 +455,26 @@ def call(self, parameters):
         @param parameters: An object specifying the data,
         and possible metadata, to be processed by the endpoint.  See the
         details for those object types.
-        @type parameters: For C{name-translation}, L{NameTranslationParameters},
-        otherwise L{DocumentParameters} or L{str}
+        @type parameters: Parameters types or L{str} for document request.
+        @param paramtype: Required parameters type.
         @return: A python dictionary expressing the result of the invocation.
         """
+        if paramtype and not isinstance(parameters, paramtype):
+            raise RosetteException(
+                "incompatible",
+                "The parameters must be " + str(paramtype),
+                self)
```

Comment:
This breaks stack trace because RosetteException's  __str__ method is called which cannot concatenate an EndpointCaller to a string. Could we use self.suburl here?

Reply:
Good catch. Done.